src/core/model.js

```diff
@@ -258,6 +258,9 @@ export default class Model extends Object3D {
 
   // TODO - just set attributes, don't hold on to geometry
   setGeometry(geometry) {
+    if (!geometry) {
```

Comment:
This is a bit strange to me. The reason: this kind of policies should ideally be applied API wide. I.e: if we accept null objects in one place, we should probably do it everywhere.

Is this important to code size reduction, or can it be dropped for now?

Reply:
> it is required so Model object can be created without Geometry, right now it is assumed non null always.

Sure, but why is it required here? Why can't `Model` check for `null`? That would make more sense to me.

Reply:
Sure, moving it to Model constructor.